apps/sms/views/conversation/js/compose.js

```diff
@@ -51,12 +48,15 @@ var Compose = (function() {
     lastScrollPosition: 0,
     resizing: false,
 
-    // Stop further input because the max size is exceeded
-    locked: false,
-
     // 'sms' or 'mms'
     type: 'sms',
 
+    lock: {
+      send: false,
+      edit: false,
+      type: null
+    },
```

Comment:
once each time we enter the view (so likely in beforeEnter ?)

The name `setupLock` is fine too, if you want to keep it